src/redux/profile/profileSlice.ts

```diff
@@ -0,0 +1,32 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+
+import { Identifier } from '@/types/types';
+
+export interface ProfileState {
+  uiPanelVisibility: {
+    // The key here should match the id of the menu item,
+    // else will be ignored
+    [key: string]: boolean;
+  };
+  markNodes: Identifier[];
+}
+
+const initialState: ProfileState = {
+  uiPanelVisibility: {},
+  markNodes: []
+};
```

Comment:
If we store the uipanel visibility here, shouldn't we set it also? If we don't store it I think there is no need to have a state for it?

Reply:
Good question. I just copied over the state data structure from the middleware file... 

But looks like the top object rather describes the data structure that we get from the profile topic?  So, we should split it up into something like: 

```
export interface ProfileData {
  uiPanelVisibility: {
    // The key here should match the id of the menu item,
    // else will be ignored
    [key: string]: boolean;
  };
  markNodes: Identifier[];
}

export interface ProfileState {
  markNodes: Identifier[];
}


const initialState: ProfileState = {
  markNodes: []
};
```
but ofc move the `ProfileData` interface to the middleware file.  Do I understand your implementation correctly then, @ylvaselling ?